tests_backend/tests/cli/test_cli.ts

```diff
@@ -0,0 +1,203 @@
+import { exists } from 'https://deno.land/std@0.202.0/fs/mod.ts'
+import {
+  mergeReadableStreams,
+} from 'https://deno.land/std@0.201.0/streams/merge_readable_streams.ts'
+import { BlobReader, TextWriter, ZipReader } from 'https://deno.land/x/zipjs/index.js'
+import { type RunnableTest, type SupabaseType, assert } from '../../utils.ts'
+import { getSupabaseSecret, getUpdateBaseData, responseOk, sendUpdate, testPlaywright } from '../../utils.ts'
+
+let cliPath: string | null = null
+let appPath: string | null = null
+const semver = `1.0.${Date.now()}`
+
+// This comes from seed.sql
+const defaultApiKey = 'ae6e7458-c46d-4c00-aa3b-153b0b8520ea'
+const defaultPackageJson = `{
+  "name": "test-cli-app",
+  "version": "1.0.0",
+  "description": "An Amazing Test App",
+  "dependencies": {
+  },
+  "devDependencies": {
+    "@capacitor/cli": "^5.4.1",
+    "typescript": "^5.2.2"
+  },
+  "author": ""
+}`
+const indexJsCode
+= `console.log('Hello world!!!');
+notifyAppReady();\n`
+
+function generateDefaultJsonCliConfig(baseUrl: URL) {
+  return {
+    appId: 'com.demo.app',
+    appName: 'demoApp',
+    webDir: 'dist',
+    bundledWebRuntime: false,
+    plugins: {
+      CapacitorUpdater: {
+        autoUpdate: true,
+        localS3: true,
+        localHost: 'http://localhost:5173',
+        localWebHost: 'http://localhost:5173',
+        localSupa: `http://${baseUrl.host}`,
+        localSupaAnon: getSupabaseSecret(),
+        statsUrl: new URL('stats', baseUrl).toString(),
+        channelUrl: new URL('channel_self', baseUrl).toString(),
+        updateUrl: new URL('updates', baseUrl).toString(),
+      },
+    },
+  }
+}
+
```

Comment:
Help me understand your design here.

Why not re-use code from the [demo-app](https://github.com/Cap-go/demo-app) by creating a separate branch that contains the code for what you are trying to do here? 

Since you are already spawning subprocesses, I'm guessing it should be straight forward to `git clone` your branch of the repo then invoke the demo-app from your tests, right?

Of course, you don't have to redo your code but I just want to better understand why you took this approach.

Reply:
> Why not re-use code from the [demo-app](https://github.com/Cap-go/demo-app)

I originally wanted to clone the `demo-app`, even asked martin about it but then I thought that that it would be wasteful as cloning it and then running `pnpm install` would install a lot of dependencies we do not need. Also this bundle is never used in any device so building this demo app would add extra unnecessary cost (in developers time and in the CI/CD execution time)

I have wrote code that generates a minimal project that can be uploaded without requiring any unnecessary  dependencies. This `generateDefaultJsonCliConfig` has to be here to not hard code any values and although I could have just copied `index.js` and `package.json` I thought it was easier to generate it in code instead of copying a file.

If there ever is a test that will start android emulator with the newly uploaded bundle then I think we should clone `demo-app` but right now it is not needed

Reply:
@WcaleNieWolny why you are not using capgo itself as test upload ?
Capgo is a mobile app as well look the file capacitor.config.ts, it uses Capgo cloud to self update.